File: practices/web_cache/web.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    vector<int> mxl;
    int miss = 0;

    for (int i = 0; i < m; i++) {
        int cac;
        cin >> cac;

        if (find(mxl.begin(), mxl.end(), cac) != mxl.end()) {
            continue;
        }
        else {
            if (mxl.size() < n) {
                mxl.push_back(cac);
                miss++;
            } else {
                mxl.erase(mxl.begin());
                mxl.push_back(cac);
                miss++;
            }
        }
    }

    cout << miss;
    return 0;
}

File: practices/web_cache/web.py
nm = input()
n, m = nm.split()
n = int(n)
m = int(m)

mxl = []
miss = 0

for i in range(m):
    cac = int(input())

    if cac in mxl:
        pass
    else:
        if len(mxl) < n:
            mxl.append(cac)
            miss += 1
        else:
            mxl.pop(0)
            mxl.append(cac)
            miss += 1

print(miss)